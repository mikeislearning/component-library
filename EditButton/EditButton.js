import React from 'react';
import { Image, Text, TouchableOpacity } from 'react-native';
import { PropTypes } from 'prop-types';
import { images } from 'app/theme';

import styles from './EditButton.styles';

export default class EditButton extends React.Component {
  static propTypes = {
    accessibilityLabel: PropTypes.string,
    icon: PropTypes.string,
    style: PropTypes.object,
    text: PropTypes.string,
    onPress: PropTypes.func.isRequired,
  };

  static defaultProps = {
    icon: 'edit',
    text: 'Edit',
  };

  render() {
    const { accessibilityLabel, icon, style, text, onPress } = this.props;
    const containerStyle = [styles.container];
    if (style) containerStyle.push(style);

    return (
      <TouchableOpacity
        accessibilityLabel={accessibilityLabel}
        activeOpacity={0.7}
        onPress={onPress}
        style={containerStyle}
      >
        <Image source={images[icon]} style={styles.editIcon} />
        <Text style={styles.text}>{text}</Text>
      </TouchableOpacity>
    );
  }
}
