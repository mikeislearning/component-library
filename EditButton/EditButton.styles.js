import { StyleSheet } from 'react-native';
import { colors, fonts } from 'app/theme';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  editIcon: {
    height: 10,
    marginRight: 5,
    marginTop: 2,
    width: 10,
  },
  text: {
    color: colors.blue,
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: fonts.size.small,
  },
});

export default styles;
