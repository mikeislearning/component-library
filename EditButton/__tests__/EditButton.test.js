import React from 'react';
import { fireEvent, render } from 'react-native-testing-library';

import EditButton from '../EditButton';

describe('<EditButton />', () => {
  it('renders correctly', () => {
    const button = render(<EditButton onPress={() => {}} />);
    expect(button).toMatchSnapshot();
  });

  it('can be pressed successfully', () => {
    const mockedFn = jest.fn();
    const { getByText } = render(<EditButton onPress={mockedFn} />);
    fireEvent.press(getByText('Edit'));
    expect(mockedFn).toHaveBeenCalledTimes(1);
  });
});
