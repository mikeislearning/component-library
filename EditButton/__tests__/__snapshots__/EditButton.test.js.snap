// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditButton /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/images/edit.png",
      }
    }
    style={
      Object {
        "height": 10,
        "marginRight": 5,
        "marginTop": 2,
        "width": 10,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#4145C8",
        "fontFamily": "MessinaSans-Regular",
        "fontSize": 12,
      }
    }
  >
    Edit
  </Text>
</View>
`;
